.github/actions/python-setup/action.yml

```diff
@@ -0,0 +1,21 @@
+name: 'Python Setup'
+description: 'Install uv, python and dependencies'
+
+runs:
+  using: composite
+  steps:
+    - name: Install python
+      uses: actions/setup-python@v5
```

Comment:
The uv docs suggest this action will manage caching better an maybe be faster than using the uv managed python. I guess this might change in the future.